### DISCUSSION TITLE
502 Bad Gateway when running behind AWS ALB

### DISCUSSION THREAD

**Discussion by deleugpn** [php/frankenphp#681] [Q&A] [answered]
I built a hello world container to test it out in the hopes of using FrankenPHP in production, but I'm stuck with no clear ideas on how to fix this.

I have a container running on AWS Fargate. For debugging purposes, I placed it on a public subnet with auto-assign IP. When I use the Public IP, the webserver works flawlessly and I get the welcome page from Laravel (HTTP-only).

When I use a DNS, it routes to Route 53, which points to an Application Load Balancer. The ALB has an HTTPS Listener that performs SSL termination and matches a Listener Rule for my Target Group using the Host Header.
When I register my container on the Target Group, I get 502 Bad Gateway from the container. The Healthcheck on the Target Group also fails. It seems that whatever Caddy is doing works perfectly with direct connection, but when going through an ALB, Caddy's response is not being properly handled by AWS.

Any suggestions would be welcomed.

Additional information that may be relevant.

Dockerfile:

```
FROM dunglas/frankenphp:1-php8.3

WORKDIR /tmp

RUN apt update \
&&  apt install supervisor redis procps -y

# Composer
RUN curl -sS https://getcomposer.org/installer | php -- --install-dir=/usr/local/bin --filename=composer

# PHP Extensions
RUN install-php-extensions redis opcache pcntl

RUN cp /usr/local/etc/php/php.ini-production /usr/local/etc/php/php.ini

WORKDIR /app

COPY . /app

RUN composer install --no-dev --optimize-autoloader

CMD ["/usr/bin/supervisord", "-n"]
```

Supervisord configuration:
```
[program:web]
process_name=%(program_name)s_%(process_num)02d
command=/usr/local/bin/frankenphp php-server --root /app/public
autostart=true
autorestart=true
stopasgroup=true
numprocs=1
redirect_stderr=true
stdout_logfile=/dev/stdout
stdout_logfile_maxbytes = 0
stopwaitsecs=3600
```

**Comment by dunglas** [comment]
Can you try to set the `SERVER_NAME` env var to `:80` to disable HTTPS?

**Reply by deleugpn** [reply, marked as answer]
I had tried that, it didn't change anything, but I found the issue: https://blog.deleu.dev/frankenphp-502-bad-gateway-behind-aws-alb/
